apps/prairielearn/src/middlewares/logResponse.js

```diff
@@ -24,39 +24,23 @@ module.exports = function (req, res, next) {
         path,
         params: req.params,
         body: req.body,
-        authn_user_id: res.locals && res.locals.authn_user ? res.locals.authn_user.user_id : null,
-        authn_user_uid: res.locals && res.locals.authn_user ? res.locals.authn_user.uid : null,
-        user_id: res.locals && res.locals.user ? res.locals.user.user_id : null,
-        user_uid: res.locals && res.locals.user ? res.locals.user.uid : null,
-        course_id: res.locals && res.locals.course ? res.locals.course.id : null,
-        course_short_name: res.locals && res.locals.course ? res.locals.course.short_name : null,
-        course_instance_id:
-          res.locals && res.locals.course_instance ? res.locals.course_instance.id : null,
-        course_instance_short_name:
-          res.locals && res.locals.course_instance ? res.locals.course_instance.short_name : null,
-        assessment_id: res.locals && res.locals.assessment ? res.locals.assessment.id : null,
-        assessment_directory:
-          res.locals && res.locals.assessment ? res.locals.assessment.tid : null,
-        assessment_instance_id:
-          res.locals && res.locals.assessment_instance ? res.locals.assessment_instance.id : null,
-        question_id: res.locals && res.locals.question ? res.locals.question.id : null,
-        question_directory:
-          res.locals && res.locals.question ? res.locals.question.directory : null,
-        instance_question_id:
-          res.locals && res.locals.instance_question ? res.locals.instance_question.id : null,
+        finished: res.writableFinished,
+        authn_user_id: res.locals?.authn_user?.user_id ?? null,
+        authn_user_uid: res.locals?.authn_user?.uid ?? null,
+        user_id: res.locals?.user?.user_id ?? null,
+        user_uid: res.locals?.user?.uid ?? null,
+        course_id: res.locals?.course?.id ?? null,
+        course_short_name: res.locals?.course?.short_name ?? null,
+        course_instance_id: res.locals?.course_instance?.id ?? null,
+        course_instance_short_name: res.locals?.course_instance?.short_name ?? null,
+        assessment_id: res.locals?.assessment?.id ?? null,
+        assessment_directory: res.locals?.assessment?.tid ?? null,
+        assessment_instance_id: res.locals?.assessment_instance?.id ?? null,
+        question_id: res.locals?.question?.id ?? null,
+        question_directory: res.locals?.question?.directory ?? null,
+        instance_question_id: res.locals?.instance_question?.id ?? null,
       };
       logger.verbose('response', access);
```

Comment:
I'm wondering if we should additionally add:
```
if (!res.writableFinished) {
  logger.verbose('request aborted by client', {response_id: res.locals.response_id, timestamp: new Date().toISOString()});
}
```
This isn't adding any new information, but in the past it's been helpful when scanning the logs to be able to see when there are a lot of the old `response was not logged` messages, which typically indicated that something was going wrong somewhere. With our improved observability these days this might no longer matter, however. @nwalters512 ?

Reply:
I see that Jonatan is already logging this as a `finished` property on the log info above, but I do agree for the case where we're manually tailing logs, it's helpful to have a specific and recognizable prefix to look for (as opposed to looking for a specific key/value pair in the entire response context).

Re: observability, it's probably worth adding OTel metrics for requests/responses - think `http.requests` to track the total number of requests and `http.requests.aborted` to track the number of, well, aborted requests. That's outside the scope of this PR, of course.

Reply:
Done in 1272708f1.